src/jit/compiler.h

```diff
@@ -8106,12 +8106,14 @@ public :
         // callbacks are needed creates GT_RETURN(TYP_BYREF, op1 = Addr of RetBuf) for
         // methods with hidden RetBufArg.
         //
-        // TODO-AMD64-Unix - As per this ABI, addr of RetBuf needs to be returned by
-        // methods with hidden RetBufArg.  Right now we are special casing GT_RETURN
-        // of TYP_VOID in codegenxarch.cpp to generate "mov rax, addr of RetBuf".
-        // Instead we should consider explicitly materializing GT_RETURN of TYP_BYREF
-        // return addr of RetBuf in IR.
-        return compIsProfilerHookNeeded() && (info.compRetBuffArg != BAD_VAR_NUM);
+        // As per the System V ABI, the address of RetBuf needs to be returned by
+        // methods with hidden RetBufArg in RAX. In such case GT_RETURN is of TYP_BYREF,
+        // returning the address of RetBuf.
```

Comment:
The comments are repetitive. I think that the general idea is that there are cases where the return buffer needs to be explicitly returned. You should say that first, and then enumerate the cases, since they both result in the same IR.